payload.go

```diff
@@ -102,8 +105,8 @@ func (ph *payloadHeader) ReadFromPayload() error {
 }
 
 // NewPayload creates a new Payload struct
-func NewPayload(filename string) Payload {
-	payload := Payload{
+func NewPayload(filename string) *Payload {
+	payload := &Payload{
```

Comment:
this is done to:
- unify usage of `Payload` across the project (since it's used only as a pointer)
- fix the underlying issue of copying the mutex